File: array/896_monotonic_array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        positive = True
        negative = True
        for i in range(len(nums) - 1):
            positive &= nums[i] <= nums[i+1]
            negative &= nums[i] >= nums[i+1]
        return positive or negative 
        

File: array/896_monotonic_array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        if (nums.length <= 2) return true;

        boolean in = true;
        boolean de = true;

        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) in = false;
            if (nums[i] < nums[i + 1]) de = false;
        }
        return in || de;
    }
}